src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -38,6 +45,24 @@ class DateTimeNormalizer implements NormalizerInterface, DenormalizerInterface,
 
     public function __construct(array $defaultContext = [])
     {
+        $this->defaultContext = [
+            self::FORMAT_KEY => \DateTime::RFC3339,
+            self::TIMEZONE_KEY => null,
+        ];
+
+        if (!\is_array($defaultContext)) {
+            @trigger_error('Passing the date time format directly to the constructor is deprecated since Symfony 4.2, use the default context instead.', E_USER_DEPRECATED);
+
+            $defaultContext = [self::FORMAT_KEY => (string) $defaultContext];
+            $defaultContext[self::TIMEZONE_KEY] = $timezone;
+        }
+
+        if (!isset($defaultContext[self::TIMEZONE_KEY]) && null !== $timezone) {
```

Comment:
There is no `$timezone` variable in the constructor.

Reply:
There was a $timezone and $defaultContext was not always an array when this was written.